crates/solver/src/driver.rs

```diff
@@ -463,8 +464,25 @@ impl Driver {
                 .solutions
                 .iter()
                 .map(|solution| solution.solver_address)
-                .collect::<HashSet<_>>() // to avoid duplicates
-                .into_iter()
+                .collect::<HashSet<_>>(); // to avoid duplicates
+                                          // external prices for all tokens contained in the trades of a settlement
+            let prices = winning_settlement
+                .settlement
+                .trades()
+                .flat_map(|trade| {
+                    let sell_token = trade.order.data.sell_token;
+                    let buy_token = trade.order.data.buy_token;
+                    vec![
+                        (
+                            sell_token,
+                            auction_prices.get(&sell_token).cloned().unwrap_or_default(),
```

Comment:
Is unwrap_or_default correct? I feel like either it is expected that we have prices for all tokens, in which case this should be plain unwrap, or we shouldn't insert 0 prices for tokens we don't have a price for.

Reply:
We discussed it here https://github.com/cowprotocol/services/pull/1282#discussion_r1119313176